dev/src/bundle.ts

```diff
@@ -0,0 +1,208 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {firestore, google} from '../protos/firestore_v1_proto_api';
+
+import {DocumentSnapshot} from './document';
+import {QuerySnapshot} from './reference';
+import {Timestamp} from './timestamp';
+import {
+  invalidArgumentMessage,
+  validateMaxNumberOfArguments,
+  validateMinNumberOfArguments,
+  validateString,
+} from './validate';
+
+import api = google.firestore.v1;
+
+const BUNDLE_VERSION = 1;
+
+/**
+ * Builds a Firestore data bundle with results from the given document and query snapshots.
+ */
```

Comment:
Nit: Add `@private` for now to hide from JSDoc.

Reply:
Done.